packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -347,21 +347,30 @@ export class AskarWallet implements Wallet {
    * Create a key with an optional seed and keyType.
    * The keypair is also automatically stored in the wallet afterwards
    *
-   * @param seed string The seed for creating a key
+   * @param secretKey string Optional secret for creating a key
+   * @param seed string Optional seed for creating a key
    * @param keyType KeyType the type of key that should be created
    *
    * @returns a Key instance with a publicKeyBase58
    *
    * @throws {WalletError} When an unsupported keytype is requested
    * @throws {WalletError} When the key could not be created
    */
-  public async createKey({ seed, keyType }: WalletCreateKeyOptions): Promise<Key> {
+  public async createKey({ seed, secretKey, keyType }: AskarWalletCreateKeyOptions): Promise<Key> {
     try {
       if (keyTypeSupportedByAskar(keyType)) {
         const algorithm = keyAlgFromString(keyType)
 
-        // Create key from seed
-        const key = seed ? AskarKey.fromSeed({ seed: Buffer.from(seed), algorithm }) : AskarKey.generate(algorithm)
+        if (seed && secretKey) {
+          throw new AriesFrameworkError('Only one of seed and secretKey can be set')
+        }
+
+        // Create key
+        const key = secretKey
+          ? AskarKey.fromSecretBytes({ secretKey, algorithm })
+          : seed
+          ? AskarKey.fromSeed({ seed: Buffer.from(seed), algorithm })
```

Comment:
Now we know what´s the actual behaviour of indy-sdk, I think it will be better to name the things as they really are. So I´m OK on adding this secretKey (being a Key or Buffer) to the geneic `WalletCreateKeyOptions`. 

But are we going to support both seed and secretKey in the interface and throw error when a non supported parameter is specified? For instance, if we specify `seed` for a IndyWallet we know it will not be treated as a seed, so should we reject it and leverage Indy users to provide a `secretKey` instead?

And what about the fallback to SigningProviders? Current interface only supports `seed`. I don´t see a limitation to add `secretKey` as an input there, but we´ll need to implement it.

There is also another interface using `seed` incorrectly: `DidConfig`, which we are using in initPublicDid method, that will be deprecated soon so I guess we should not care about it.

Reply:
Yeah good points. I thinkg throwing an error in the indy-sdk seems fine. Then in Askar we can implement both and we could also add the secretKey/seed paramteres to the signing provider.

One thing we should look at is that it used `privateKey` there, and we're using secretKey here. I'm more used to `privateKey`, but if we go for `secretKey` we may want to make that consistent across the framework.

Reply:
Also, we could also call it importKey or something as you're not really creating a key when you create it based on the secret bytes

Reply:
I think for the moment we can go with an optional `privateKey` (as Uint8Array) for WalletCreateKeyOptions and throw error if using seed in IndyWallet, and throw error if using privateKey with signing providers (I didn't look a lot but didn't find a way to provide a private key in bbs-signatures).

Reply:
If we create it from private key, we already have the private key rigt, so it should work with the signing providers I'd assume?

Reply:
I think so. We can make `CreateKeyPairOptions` interface from  `SigningProvider` to accept both (in the same way Wallet does now) and then depending on each implementation it will be supported or not.